### PR TITLE
Require Audio Easing support on RialtoMSEAudioSink for Netflix (fade-volume prop) 

### PR DESCRIPTION
**Pull request by skyshaha1**
Summary: Require Audio Easing support on RialtoMSEAudioSink for Netflix - fade volume prop
Type: Fix
Test Plan: UT/ CT
Jira: RIALTO-593

### CODE REVIEW

File: media/server/gstplayer/source/GstGenericPlayer.cpp [hunk 1]
```diff
@@ -1457,16 +1457,42 @@ void GstGenericPlayer::setVolume(double targetVolume, uint32_t volumeDuration, f
...
+    GstElement *sink{getSink(MediaSourceType::AUDIO)};
+    if (!sink)
+    {
+        RIALTO_SERVER_LOG_ERROR("Failed to get fade volume property, sink is NULL");
+        return false;
+    }
+
+    if (m_glibWrapper->gObjectClassFindProperty(G_OBJECT_GET_CLASS(sink), "fade-volume"))
+    {
+        gdouble fadeVolume;
```
Comment (skyfroxl) [inline media/server/gstplayer/source/GstGenericPlayer.cpp:1478]: Isn't the fadevolume property and int? also initialise  this variable to a invalid value

File: media/server/gstplayer/source/GstGenericPlayer.cpp [hunk 2]
```diff
@@ -1457,16 +1457,42 @@ void GstGenericPlayer::setVolume(double targetVolume, uint32_t volumeDuration, f
...
     // constructor and destructor. GstGenericPlayer is created/destructed on main thread, so we won't have a crash here.
     if (!m_context.pipeline)
     {
         return false;
     }
-    volume =
-        m_gstWrapper->gstStreamVolumeGetVolume(GST_STREAM_VOLUME(m_context.pipeline), GST_STREAM_VOLUME_FORMAT_LINEAR);
+
+    GstElement *sink{getSink(MediaSourceType::AUDIO)};
+    if (!sink)
```
Comment (skyfroxl) [inline media/server/gstplayer/source/GstGenericPlayer.cpp:1470]: Sink might not be present but we can still set volume on the pipeline

File: tests/componenttests/server/tests/mediaPipeline/VolumeTest.cpp [hunk 3]
```diff
@@ -79,8 +79,15 @@ class VolumeTest : public MediaPipelineTest
...
     {
+        mockAudioSink();
+
+        EXPECT_CALL(*m_glibWrapperMock, gObjectClassFindProperty(_, StrEq("fade-volume"))).WillOnce(Return(nullptr));
+
         EXPECT_CALL(*m_gstWrapperMock, gstStreamVolumeGetVolume(_, GST_STREAM_VOLUME_FORMAT_LINEAR))
             .WillOnce(Return(kVolume));
+
+        EXPECT_CALL(*m_gstWrapperMock, gstObjectUnref(m_audioSink))
+            .WillOnce(Invoke(this, &MediaPipelineTest::workerFinished));
```
Comment (skyfroxl) [inline tests/componenttests/server/tests/mediaPipeline/VolumeTest.cpp:90]: This is not on the worker thread

File: tests/unittests/media/server/gstplayer/genericPlayer/GstGenericPlayerTest.cpp [hunk 4]
```diff
@@ -125,6 +125,16 @@ class GstGenericPlayerTest : public GstGenericPlayerTestCommon
                         *returnVal = value;
                     }));
         }
+        else if constexpr (std::is_same_v<T, double>)
+        {
+            EXPECT_CALL(*m_glibWrapperMock, gObjectGetStub(_, StrEq(propertyName.c_str()), _))
```
Comment (skyfroxl) [inline tests/unittests/media/server/gstplayer/genericPlayer/GstGenericPlayerTest.cpp:130]: Wont need this if its an int

File: tests/unittests/media/server/gstplayer/genericPlayer/GstGenericPlayerTest.cpp [hunk 5]
```diff
@@ -500,20 +490,23 @@ TEST_F(GstGenericPlayerTest, shouldFailToGetStatsInPlayingStateIfStructIncomplet
     EXPECT_FALSE(m_sut->getStats(MediaSourceType::VIDEO, returnedRenderedFrames, returnedDroppedFrames));
 }
 
-TEST_F(GstGenericPlayerTest, shouldFailToGetVolumeAsNullAudioSink)
+TEST_F(GstGenericPlayerTest, shouldFailToGetFadeVolumeWhenAudioSinkIsNull)
```
Comment (skyfroxl) [inline tests/unittests/media/server/gstplayer/genericPlayer/GstGenericPlayerTest.cpp:493]: Not a failure, change too ShouldGetVolumeWhenAudioSinkIsNull

**Review by skyfroxl — APPROVED** [review]

**Review by skyproudmanp — APPROVED** [review]

**Pull request merged**